src/core/Form/Combobox/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,595 @@
+import React, { Component } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { HtmlDiv } from '../../../../reset';
+import { windowAvailable } from '../../../../utils/common';
+import { AutoId } from '../../../../utils/AutoId';
+import { Debounce } from '../../../utils/Debounce/Debounce';
+import { Button } from '../../../Button/Button';
+import { Chip } from '../../../Chip';
+import { Popover } from '../../../Popover/Popover';
+import { FilterInput, FilterInputStatus } from '../../FilterInput/FilterInput';
+import { ComboboxItemList } from '../ComboboxItemList/ComboboxItemList';
+import { ComboboxItem } from '../ComboboxItem/ComboboxItem';
+import { ComboboxEmptyItem } from '../ComboboxEmptyItem/ComboboxEmptyItem';
+import { ChipList } from '../ChipList/ChipList';
+import { baseStyles } from './Combobox.baseStyles';
+
+const baseClassName = 'fi-combobox';
+const comboboxClassNames = {
+  wrapper: `${baseClassName}_wrapper`,
+  open: `${baseClassName}--open`,
+  removeAllButton: `${baseClassName}_removeAllButton`,
+  error: `${baseClassName}--error`,
+};
+
+export interface ComboboxData {
+  /** Unique label that will be shown on combobox item and used on filter */
+  labelText: string;
+  /** Using labelText if chipText is not given */
+  chipText?: string;
+  /** Item selection disabled for the user */
+  disabled?: boolean;
+  /** Unique id to identify the item */
+  uniqueItemId: string;
+}
+
+export interface ComboboxProps<T extends ComboboxData> {
+  /** Combobox container div class name for custom styling. */
+  className?: string;
+  /** Items for the combobox */
+  items: Array<T & ComboboxData>;
+  /**
+   * Unique id
+   * If no id is specified, one will be generated automatically
+   */
+  id?: string;
+  /** Label */
+  labelText: string;
+  /** Event that is sent when item selections change */
+  onItemSelectionsChange?: (selectedItems: Array<T>) => void;
```

Comment:
would simply `onSelectionChange` be too ambiguous?

Reply:
Is this redundant as there is onItemSelected as well?

Reply:
Dropping the `item` from prop names would be okay for me, as there is no other stuff to be selected than those items.

It is true that both are touching same area, so we can make changes here if needed. 
Here are the differences:

### onItemSelectionsChange
The first `onItemSelectionsChange` provides **all** the items when selection changes. 
e.g user could store current selections somewhere with these.

### onItemSelect
And the latter, `onItemSelect`,  just the **one** that is selected, with along it's `uniqueItemId`.
e.g user can use this for controlled stuff or some other detailed way for just that specific item.

Reply:
Having both makes sense, but I'm just wondering if it's necessary? Maybe leave it as it is.